lib/stylesheet/manager.rb

```diff
@@ -86,8 +86,9 @@ def self.precompile_css
     themes = Theme.where('user_selectable OR id = ?', SiteSetting.default_theme_id).pluck(:id, :name)
     themes << nil
     themes.each do |id, name|
-      [:desktop, :mobile, :desktop_rtl, :mobile_rtl].each do |target|
+      [:desktop, :mobile, :desktop_rtl, :mobile_rtl, :desktop_theme, :mobile_theme, :admin].each do |target|
         theme_id = id || SiteSetting.default_theme_id
+        next if target =~ THEME_REGEX && theme_id == -1
```

Comment:
This line is a safeguard to prevent the precompile process from exploding on sites that don't have a theme set as the site's default theme. On sites that have no default theme (i.e. sites that have `SiteSetting.default_theme_id == -1`), there is no theme CSS we need to serve with core's CSS, thus there is no theme CSS that we need to precompile. So I'm just skipping precompiling `desktop_theme` and `mobile_theme` here because if we try it's going to explode. Does that make sense?